### PR TITLE
Update project to standards

### PR DESCRIPTION
**Pull request by JasonBock**
## Description
Updating ObjectFactory to supported frameworks and standards

### CODE REVIEW

File: RockLib.Configuration.ObjectFactory/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,68 @@
+root = true
+
+[*]
+end_of_line = crlf
```
Comment (eslutz) [inline RockLib.Configuration.ObjectFactory/.editorconfig]: Yesterday @NeilSchiefer brought up that we should change this from `crlf` to `lf` to keep everything consistent with what's already in place, so I'm not sure if you want to go ahead and update it here.

File: RockLib.Configuration.ObjectFactory/ConfigReloadingProxy.cs [hunk 2]
```diff
@@ -9,198 +9,211 @@
...
-            {
-                // The value section depends on whether the 'ReloadOnChange' flag is set to true.
-                if (string.Equals(_section[ConfigurationObjectFactory.ReloadOnChangeKey]?.ToLowerInvariant(), "true"))
-                    valueSection = _section.GetSection(ConfigurationObjectFactory.ValueKey);
-                else
-                    valueSection = _section;
-            }
-
-            // Throw if no type can be located.
+   // Given that this class is already public,
```
Comment (NeilSchiefer) [inline RockLib.Configuration.ObjectFactory/ConfigReloadingProxy.cs:12]: This is what I was saying with the end_of_line .editorconfig setting.  It makes it hard to identify the real changes. We should probably change it to just lf.  Or go full on changing to crllf but then we should probably change every file in the repo in a separate pr.

File: RockLib.Configuration.ObjectFactory/ConfigReloadingProxy.cs [hunk 3]
```diff
@@ -9,198 +9,211 @@
...
+      private TInterface CreateObject()
+      {
+         // In order to create the object (and avoid infinite recursion), we need to figure out
+         // the concrete type to create and the config section that defines the value to create.
+         Type? concreteType;
+         IConfiguration valueSection;
+
+         // If _section contains a type-specified value, extract the type and use the value sub-section.
+         string typeValue = _section[ConfigurationObjectFactory.TypeKey];
+         if (typeValue != null)
```
Comment (eslutz) [inline RockLib.Configuration.ObjectFactory/ConfigReloadingProxy.cs]: It looks like in some places == null or != null was updated to is null or is not null, but it looks like some might have been missed.

File: RockLib.Configuration.ObjectFactory/ConfigReloadingProxyFactory.cs [hunk 4]
```diff
@@ -11,395 +11,409 @@
...
-            return moduleBuilder.DefineType(name, ProxyClassAttributes, baseType, new[] { interfaceType });
-        }
-
-        private static void AddConstructor(TypeBuilder proxyTypeBuilder, ConstructorInfo baseConstructor)
-        {
-            var constructorBuilder = proxyTypeBuilder.DefineConstructor(ConstructorAttributes, baseConstructor.CallingConvention,
-                new[] { typeof(IConfiguration), typeof(DefaultTypes), typeof(ValueConverters), typeof(Type), typeof(string), typeof(IResolver) });
-
-            var il = constructorBuilder.GetILGenerator();
+      // TODO: Can these really be null?
```
Comment (eslutz) [inline RockLib.Configuration.ObjectFactory/ConfigReloadingProxyFactory.cs]: Is this just a note for the next time we go through the libraries, or did you mean to do something with this?

File: RockLib.Configuration.ObjectFactory/Resolver.cs [hunk 5]
```diff
@@ -1,206 +1,187 @@
...
+            if (canResolveNamed(p.ParameterType, p.Name!))
+               return true;
+
+            return canResolve(p.ParameterType);
+         };
+
+         Resolve = p =>
+         {
+            var obj = resolveNamed(p.ParameterType, p.Name!);
+            if (obj != null)
```
Comment (eslutz) [inline RockLib.Configuration.ObjectFactory/Resolver.cs]: Can be changed from != null to is not null.

File: Tests/RockLib.Configuration.ObjectFactory.Tests/.editorconfig [hunk 6]
```diff
@@ -0,0 +1,68 @@
+root = true
+
+[*]
+end_of_line = crlf
```
Comment (eslutz) [inline Tests/RockLib.Configuration.ObjectFactory.Tests/.editorconfig]: Same as the other project.  This should probably be changed to `lf`.

File: RockLib.Configuration.ObjectFactory/.editorconfig (hunk 1 above)
Comment (JasonBock) [reply RockLib.Configuration.ObjectFactory/.editorconfig]: I'm fine with this, if the `.editorconfig` copy in our assets repo hasn't been updated, it should be as that's what we should be using to start with.

File: RockLib.Configuration.ObjectFactory/ConfigReloadingProxy.cs (hunk 2 above)
Comment (JasonBock) [reply RockLib.Configuration.ObjectFactory/ConfigReloadingProxy.cs:12]: See my previous comment. If we want `lf`, I'm fine with that, we just need our "base" copy to reflect that.

File: RockLib.Configuration.ObjectFactory/ConfigReloadingProxy.cs (hunk 3 above)
Comment (JasonBock) [reply RockLib.Configuration.ObjectFactory/ConfigReloadingProxy.cs]: We can use `is null`, but not `is not null`. That won't work for all target frameworks.

FWIW I just did another search and I didn't find any `== null` in the code.

File: RockLib.Configuration.ObjectFactory/ConfigReloadingProxyFactory.cs (hunk 4 above)
Comment (JasonBock) [reply RockLib.Configuration.ObjectFactory/ConfigReloadingProxyFactory.cs]: That comment was resolved, I just forgot to remove it (and now I have).

File: RockLib.Configuration.ObjectFactory/Resolver.cs (hunk 5 above)
Comment (JasonBock) [reply RockLib.Configuration.ObjectFactory/Resolver.cs]: OK....I retract my previous comment. Apparently `is not null` *will* work with all frameworks. I'll triple-check again, but if it does work, then I'll update them all.

File: Tests/RockLib.Configuration.ObjectFactory.Tests/.editorconfig (hunk 6 above)
Comment (JasonBock) [reply Tests/RockLib.Configuration.ObjectFactory.Tests/.editorconfig]: Done

File: RockLib.Configuration.ObjectFactory/.editorconfig (hunk 1 above)
Comment (eslutz) [reply RockLib.Configuration.ObjectFactory/.editorconfig]: @NeilSchiefer put in a PR to update it yesterday after we talked about it.  I approved it, but it hadn't been merged yet, so I'll take care of that.

**Review by eslutz — APPROVED** [review]

**Pull request merged**